custom_components/wattbox/button.py

```diff
@@ -0,0 +1,117 @@
+"""Button platform for wattbox."""
+
+import logging
+import re
+from typing import Any, List
+
+from homeassistant.components.button import ButtonDeviceClass, ButtonEntity
+from homeassistant.const import CONF_NAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from pywattbox.base import BaseWattBox, Outlet
+
+from .const import (
+    CONF_NAME_REGEXP,
+    CONF_SKIP_REGEXP,
+    DOMAIN_DATA,
+    RESTART_ICON,
+)
+from .entity import WattBoxEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_regex(config: ConfigType, key: str) -> re.Pattern[str] | None:
+    regexp_str: str = config.get(key, "")
+    if regexp_str:
+        try:
+            return re.compile(regexp_str)
+        except re.error:
+            _LOGGER.error("Invalid %s: %s", key, regexp_str)
+    return None
```

Comment:
If not deduping `switch` and `button` entirely, should probably put this somewhere shared so they can both import the same function.